### PR TITLE
luci-app-acme: `cfgvalue` does not receive the `set_value` parameter when it is called in `CBIValue.render`.

### PR DESCRIPTION
**Pull request by tsanie**
- [X] This PR is not from my *main* or *master* branch :poop:, but a *separate* branch :white_check_mark:
- [X] Each commit has a valid :black_nib: `Signed-off-by: <my@email.address>` row (via `git commit --signoff`)
- [X] Each commit and PR title has a valid :memo: `<package name>: title` first line subject for packages
- [ ] Incremented :up: any `PKG_VERSION` in the Makefile
- [X] Tested on: (x86_64, v24.10-SNAPSHOT, Edge 132.0.2957.11) :white_check_mark:
- [ ] \( Preferred ) Mention: @ the original code author for feedback
- [ ] \( Preferred ) Screenshot or mp4 of changes:
- [ ] \( Optional ) Closes: e.g. openwrt/luci#issue-number
- [ ] \( Optional ) Depends on: e.g. openwrt/packages#pr-number in sister repo
- [X] Description: The value of "Key type" will be overwritten to `undefined` when the drop-box after rendering. (except the first migration from the old `keylength` config parameter)

### CODE REVIEW

File: applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js [hunk 1]
```diff
@@ -497,7 +502,7 @@ return view.extend({
 					default: return ''; // bad value
 				}
 			}
-			return set_value;
```
Comment (tsanie) [inline applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:500]: As code https://github.com/openwrt/luci/blob/b1968d9cdd6425821553138e0487df8840785afa/modules/luci-base/htdocs/luci-static/resources/form.js#L3681
the `set_value` will be `undefined`

**Comment by tohojo** [comment]
I'm sorry, but I only have a very limited grasp of how all the callback stuff in luci works, so I don't feel qualified to review this. Please find a luci expert to take a look...

File: applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js (hunk 1 above)
Comment (systemcrash) [reply applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:500]: your code branch simplifies to:
```
if(set_value){
 something = set_value;
}
...
return something;
```

which is the same as:
```
if(set_value) return set_value;
return null;
```

How does this differ? What is the error you receive?

Comment (tsanie) [reply applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:500]: Yes, that’s the problem.

The `cfgvalue` function should always return the selected value which isn’t `undefined`, so we need to cache it and return it later. (like the super method does) https://github.com/openwrt/luci/blob/b1968d9cdd6425821553138e0487df8840785afa/modules/luci-base/htdocs/luci-static/resources/form.js#L1826

The `set_value` will be passed in the `load` function.https://github.com/openwrt/luci/blob/b1968d9cdd6425821553138e0487df8840785afa/modules/luci-base/htdocs/luci-static/resources/form.js#L913
But in the later `render` function, the second parameter is undefined.https://github.com/openwrt/luci/blob/b1968d9cdd6425821553138e0487df8840785afa/modules/luci-base/htdocs/luci-static/resources/form.js#L3681

Issue:
Now the dropdown box always shows the undefined value even I have the right config `ec256`.
image

Comment (tsanie) [reply applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js:500]: Got it, the `super` method is a good way to resolve it.

**Pull request closed without merging**